### ISSUE TITLE
[iOS] - `dolly-v2-3b` latency + accuracy

### ISSUE THREAD

**Issue by terryworona** [mlc-ai/mlc-llm#155]
Hey all, 

## TLDR

I managed to build and deploy the `dolly-v2-3b` model to iOS `iPad Pro M1` (thanks to the helpful suggestions in #129 and #116). However, I noticed that both the _latency_ and _accuracy_ of the model is quite poor (see below). 

_Why Dolly?_ I want to use a  model open to commercial use. If there are any alternatives beyond Dolly that would work with this library, I'm all ears :) 

## Observations

1. **Latency** is quite long - it can take several minutes to stream a response. _Is this typical for an M1 device?_
2. Most output is **inaccurate** - see attachment of a few simple prompts. I find that occasionally the answer is correct, but it's surrounded by a bunch of irrelevant/erroneous content. The model never seems to stop streaming - most prompts go on forever - oftentimes in loops (see second example, where it repeatedly gets stuck streaming "_mix, mix, mix..._"). _Is this the expected performance of Dolly, or is something somehow misconfigured?_ (See: Appendix below). 

<img width="1044" alt="238210455-96bcfa11-4f1c-461b-b75d-42fe7379f985" src="https://github.com/mlc-ai/mlc-llm/assets/358709/996c961d-d618-4fb4-8d64-3eb99afd99b7">

## Appendix

Setup instructions below:

### Download Model
```
$ git lfs install && git clone https://huggingface.co/databricks/dolly-v2-3b
```

### Link Model
```
$ mkdir -p dist/models 
$ ln -s /Users/me/Desktop/dolly-v2-3b dist/models/dolly-v2-3b 
```

### Build Model
```
$ python3 build.py --model dolly-v2-3b --dtype float16 --target iphone --quantization-mode int3 --quantization-sym --quantization-storage-nbit 16 --max-seq-len 768
```

### Mobile Integration

```
$ ./prepare_libs.sh
$ ./prepare_params.sh
```

I had to update both scripts to point to `dolly-v2-3b` (previously hardcoded to `vicuna-v1-7b`). I also had to update `prepare_params` to move Dolly's `tokenizer.json` (instead of `.model`) and update the mobile code base:

```
 std::string tokenizer_path = bundle_path + "/dist/tokenizer.json"; 
```
I also noticed via `LLMChat.mm`, that the `model` and `conv_template` properties are hardcoded to `vircuna` and `vicuna_v1.1` respectively:

```
std::string model = "vircuna";
std::string conv_template = "vicuna_v1.1";
```

_Do these matter in any way, or is it just a namespace thing? Should they be updated to point to Dolly?_

**Comment by tqchen** [comment]
close due to inactive status, we have updated support for redpajama and llama2 both support commercial use, please also checkout updated docs at https://mlc.ai/mlc-llm/docs/
